internal/zinc-core/src/main/scala/sbt/internal/inc/Stamp.scala

```diff
@@ -56,63 +57,65 @@ private[sbt] sealed abstract class StampBase extends Stamp {
 
 trait WithPattern { protected def Pattern: Regex }
 
-import java.lang.{ Long => BoxedLong }
+import java.lang.{ Long => BoxedLong, Integer => BoxedInt }
 
-/** Define the hash of the file contents. It's a typical stamp for compilation sources. */
-final class Hash private (val hexHash: String) extends StampBase {
-  // Assumes `hexHash` is a hexadecimal value.
-  override def writeStamp: String = s"hash($hexHash)"
-  override def getValueId: Int = hexHash.hashCode()
-  override def getHash: Optional[String] = Optional.of(hexHash)
+final class Hash64(val hash: Long) extends StampBase {
```

Comment:
Is it possible to just hold the hash bytes here, and use its hashCode as the hashCode? Not sure which API constraints you're working with, but: converting to 32 is necessary for JVM hashCode methods is necessary, but converting back and forth to `Long` values just shouldn't (ever) be necessary except in the text based analysis format.